Sources/Sentry/Public/SentryBaggage.h

```diff
@@ -67,6 +74,17 @@ NS_SWIFT_NAME(Baggage)
                         sampled:(nullable NSString *)sampled
                        replayId:(nullable NSString *)replayId;
 
+- (instancetype)initWithTraceId:(SentryId *)traceId
```

Comment:
As the `SentryBaggage.h` is public adding the `sampleRand` parameter could break other SDKs using sentry-cocoa.

I currently see 3 options:

1. Change the public API and potentially cause crashes, keep only one method --> cleaner
2. Add the second method with the additional parameter --> SDK users accessing this method must make sure to use the one with `sampleRand`
3. Add the second method with the additional parameter and deprecate the original one --> SDK users would get a warning, but we are treating deprecation as hard errors in Swift, so tests won't compile

Reply:
> could break other SDKs using sentry-cocoa

do you know which one's could break? e.g flutter has its own implementation for distributed tracing and react native I think inherits it from javascript, dont think this change would affect those

Reply:
Not particularly, but as the header is also in the `public` folder I am not sure who is actually able to access it. Maybe even SDK user? cc @philipphofmann

Reply:
I would go with option 3. We don't need to worry much about the deprecation because I'm sure most users don't use this class.

Public means it's public. Every user can access it.

Reply:
@philipphofmann I tried option 3 during development and calling the initializers from from Swift failed to compile because deprecation warnings are treated as compile-time errors.

I'll try to reproduce, maybe we need to remove the tests for the deprecated methods, or maybe I can somehow ignore it.

Reply:
Oh no 🤦. Maybe we should disable this warning if it gets too complicated.